src/sknnr/_gnn.py

```diff
@@ -1,12 +1,94 @@
+from __future__ import annotations
+
+from typing import Callable, Literal
+
 from sklearn.base import TransformerMixin
+from sklearn.metrics import DistanceMetric
 
 from ._base import TransformedKNeighborsRegressor, YFitMixin
 from .transformers import CCATransformer
 
 
 class GNNRegressor(YFitMixin, TransformedKNeighborsRegressor):
-    def __init__(self, n_components=None, **kwargs):
-        super().__init__(**kwargs)
+    """
+    Regression using Gradient Nearest Neighbor (GNN) imputation.
+
+    The target is predicted by local interpolation of the targets associated with
+    the nearest neighbors in the training set, calculated in transformed Canonical
+    Correspondence Analysis (CCA) space.
+
+    See `sklearn.neighbors.KNeighborsRegressor` for more information on parameters
+    and implementation.
```

Comment:
Do you think it's good as-is, or should we lean more heavily on referencing `KNeighborsRegressor`? Currently I just omitted some specific details like how each of the `algorithm` options works, but otherwise this largely duplicates the docstring.

Reply:
My feeling is that it is good as-is,